src/model_theory/encoding.lean

```diff
@@ -130,5 +168,148 @@ small_of_injective list_encode_injective
 
 end term
 
+namespace bounded_formula
+
+/-- Encodes a bounded formula as a list of symbols. -/
+def list_encode : ∀ {n : ℕ}, L.bounded_formula α n →
+  list ((Σ k, L.term (α ⊕ fin k)) ⊕ (Σ n, L.relations n) ⊕ ℕ)
+| n falsum := [sum.inr (sum.inr (n + 3))]
+| n (equal t₁ t₂) := [sum.inr (sum.inr 0), sum.inl ⟨_, t₁⟩, sum.inl ⟨_, t₂⟩]
```

Comment:
```suggestion
| n (equal t₁ t₂) := [sum.inl ⟨_, t₁⟩, sum.inl ⟨_, t₂⟩]
```
This requires only trivial changes in `list_decode` and `list_decode_encode_list`.